cmd/kube-rbac-proxy/app/kube-rbac-proxy.go

```diff
@@ -212,7 +212,15 @@ func Run(cfg *server.KubeRBACProxyConfig) error {
 		return fmt.Errorf("failed to setup an authorizer: %v", err)
 	}
 
-	proxy := httputil.NewSingleHostReverseProxy(cfg.KubeRBACProxyInfo.UpstreamURL)
+	proxy := &httputil.ReverseProxy{
+		Rewrite: func(pr *httputil.ProxyRequest) {
+			target := cfg.KubeRBACProxyInfo.UpstreamURL
+			pr.SetURL(target)
+			pr.Out.Host = target.Host
+			copyHeaderIfSet(pr.In, pr.Out, "X-Forwarded-For")
```

Comment:
Wouldn't it make sense to add the IP address of the client, if not already set?

https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/X-Forwarded-For#syntax

Reply:
`pr.SetXForwarded()` does that but only if `X-Forwarded-For` is already set

Reply:
oh, yes. I see. But why do we need to copy the `X-Forwarded-For` header? Does it get removed otherwise?

Reply:
See https://cs.opensource.google/go/go/+/refs/tags/go1.23.3:src/net/http/httputil/reverseproxy.go;l=417-424
and
https://cs.opensource.google/go/go/+/refs/tags/go1.23.3:src/net/http/httputil/reverseproxy.go;l=78-95

Reply:
Added unit tests.